### ISSUE TITLE
Unable to converge loss function while training

### ISSUE THREAD

**Issue by porwalnaman01** [nianticlabs/monodepth2#368]
Hi there. I loved reading your paper and code is very simple and easy to understand. But I am facing some difficulty while training the model. The loss is always fluctuating between 0.17 to 0.14  and not converging while training even after completing 30 epochs on raw kitti dataset. I don't understand why is this happening. I will be glad if you can help me here. Thanks in advance!

I am using RTX 3090 gpu, CUDA 11.1, PyTorch 1.8.0, Python 3.6.6, UBUNTU 18.04
command I used to train - CUDA_VISIBLE_DEVICES=7 python train.py --model_name mono_model_final --num_epochs 30

This is the training loss till the 3rd epoch, though it was still between 0.14-0.17 after the 30th epoch as well!!

"
Training model named:
   mono_model_final
Models and tensorboard events files are saved to:
   /home/dan/tmp
Training is using:
   cuda
Using split:
   eigen_zhou
There are 39810 training items and 4424 validation items

Training
/disk1/dan/zmiaa/anaconda3/envs/monodepth2/lib/python3.6/site-packages/torch/nn/functional.py:3829: UserWarning: Default grid_sample and affine_grid behavior has changed to align_corners=False since 1.3.0. Please specify align_corners=True if the old behavior is desired. See the documentation of grid_sample for details.
  "Default grid_sample and affine_grid behavior has changed "
epoch   0 | batch      0 | examples/s:   1.9 | loss: 0.16496 | time elapsed: 00h00m11s | time left: 00h00m00s
epoch   0 | batch    250 | examples/s:  27.7 | loss: 0.13513 | time elapsed: 00h01m59s | time left: 13h09m12s
epoch   0 | batch    500 | examples/s:  33.1 | loss: 0.13025 | time elapsed: 00h03m46s | time left: 12h26m43s
epoch   0 | batch    750 | examples/s:  33.1 | loss: 0.15098 | time elapsed: 00h05m30s | time left: 12h05m44s
epoch   0 | batch   1000 | examples/s:  29.1 | loss: 0.17568 | time elapsed: 00h07m17s | time left: 11h58m16s
epoch   0 | batch   1250 | examples/s:  31.7 | loss: 0.13919 | time elapsed: 00h09m03s | time left: 11h51m54s
epoch   0 | batch   1500 | examples/s:  30.7 | loss: 0.14501 | time elapsed: 00h10m49s | time left: 11h47m23s
epoch   0 | batch   1750 | examples/s:  33.9 | loss: 0.14961 | time elapsed: 00h12m35s | time left: 11h43m27s
epoch   0 | batch   2000 | examples/s:  28.3 | loss: 0.15130 | time elapsed: 00h14m20s | time left: 11h39m13s
Training
epoch   1 | batch    683 | examples/s:  29.6 | loss: 0.14263 | time elapsed: 00h27m31s | time left: 10h57m23s
epoch   1 | batch   2683 | examples/s:  30.2 | loss: 0.14044 | time elapsed: 00h40m37s | time left: 10h33m02s
Training
epoch   2 | batch   1366 | examples/s:  33.0 | loss: 0.16272 | time elapsed: 00h53m50s | time left: 10h15m49s
Training
epoch   3 | batch     49 | examples/s:  28.2 | loss: 0.15169 | time elapsed: 01h07m15s | time left: 10h02m00s
epoch   3 | batch   2049 | examples/s:  33.6 | loss: 0.15662 | time elapsed: 01h20m26s | time left: 09h46m36s
"

**Comment by mdfirman** [comment]
Hello and thanks for your interest.

You say:
```
I am using RTX 3090 gpu, CUDA 11.1, PyTorch 1.8.0, Python 3.6.6, UBUNTU 18.04
```
These is not the version of pytorch we specify in the requirements – you might have better luck if you check those.

Thanks

**Comment by porwalnaman01** [comment]
Yeah! I did see the requirements in the READme file but cuda 9.1 is not available in UBUNTU 18.04 as I checked it on the NVIDIA's website. I have cuda 11.1 on a remote server and it supports Pytorch 1.8.0. Have you tried it on this version?

**Comment by mrharicot** [comment]
Hi,
When using `conda` you can install a specific version of cuda for a conda env which is separate from the system one, as long as your drivers are more recent that this verison (you can see the max cuda version supported by your driver in `nvidia-smi`).
you can then install pytorch 0.4.1 for cuda 9.2 as follows:
```bash
conda install pytorch=0.4.1 cuda92 -c pytorch
```
https://pytorch.org/get-started/previous-versions/#via-conda

**Comment by jamesheatonrdm** [comment]
Hi, can you explain why the cuda version would prevent the model from converging? As far as I am aware that shouldn't have an effect

**Comment by mrharicot** [comment]
CUDA works in mysterious ways.
jokes aside while the cuda version shouldn't matter much, pytorch comes built with different versions of cudnn which might produce very so slightly different results for some operations (usually pooling or averaging since it reduces in parallel, for which order matters).
a lot of things have changed since pytorch 0.4.1 so I would not expect the model to behave the same on a more recent version

**Comment by jamesheatonrdm** [comment]
The thing is, I am using an NVIDIA GeForce 3060 as my GPU. It appears that cuda 9 is not compatible with my device as it requires cuda 11 or greater. Is there a way to get this to run using the correct system requirements? Or will this project just simply not work for standard graphics cards?
